website/tosti/api/serializers.py

```diff
@@ -0,0 +1,45 @@
+from rest_framework import serializers
+from rest_framework.serializers import ModelSerializer, ListSerializer
+from rest_framework.utils import model_meta
+
+
+class WritableModelSerializer(serializers.ModelSerializer):
```

Comment:
While this is a cool solution, I feel like it shouldn't be necessary. 
I suppose the problem you're solving is to allow _setting_ a related model, while rendering it with a serializer.
I believe it should actually be possible to do that without this complicated stuff:
- Add the serializer as read only field (nothing special here)
- Also add a `PrimaryKeyRelatedField` with `read_only=False, write_only=True`

I haven't tried it out but that would be a simple and elegant solution. Regarding OpenAPI docs, having the two fields use the same name might be a problem, but again I haven't tried, and working around that probably wouldn't be harder than supporting the WritableModelSerializer

Reply:
I thought about this solution and I also think that what you are saying is easier to support. What we do need to do then is to include the normal field as `normal_field` and the primary key field as something like `normal_field_id`. I decided to use this approach because I thought it was more nice to use the `normal_field` for both (to an API user, I think that is more nice to use). @JobDoesburg what do you think?

Reply:
ah right, of course it really is impossible to reuse the same name for the two. Any chance it could be done with a custom field type or overriding the serializer to_internal_value?

Reply:
Good idea, I will try this!

Reply:
Although the solution you are proposing is a lot nicer, I have tried implementing this but it seems to be a lot harder than what we have right now. Because of the information that is available after the creation of the class, it is much easier to implement the functionality on the serializer level than on the field level. I propose to use this first and create an issue to look at refactoring this at a later stage.

Reply:
Okay another update: I created a [separate branch](https://github.com/KiOui/TOSTI/tree/fix/interface-changes-b) with a working solution by only using a field. It was tough but it is possible. I however still think that the solution in this PR is easier to understand and a bit cleaner. Could you also give your opinion @DeD1rk ?

Reply:
I think doing it as it is now is alright. I have some more ideas that might work (subclass PrimaryKeyRelatedField to override its to_representation, the base field should then handle writing automatically, hopefully leaving only the schema generation as a tricky thing), but it doesn't matter so much. I might try it in a bit but no need to not merge this.